src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs

```diff
@@ -423,6 +423,44 @@ SymbolChange fixChangeIfMemberIsReAdded(IDefinition item, SymbolChange change, F
             }
         }
 
+        /// <summary>
+        /// Gets the base definition for the specified symbol, if there is one, and it has an implementation
+        /// that is callable.
+        /// </summary>
+        public IReference? TryGetBaseImplementationDefinition(ISymbolInternal symbolInternal, out bool isThisAssembly)
+        {
+            isThisAssembly = false;
+            var symbol = symbolInternal.GetISymbol();
+
+            ISymbol? baseSymbol;
+            if (symbol is IMethodSymbol method)
+            {
+                baseSymbol = method.OverriddenMethod;
+            }
+            else if (symbol is IPropertySymbol property)
+            {
+                baseSymbol = property.OverriddenProperty;
+            }
+            else if (symbol is IEventSymbol @event)
+            {
+                baseSymbol = @event.OverriddenEvent;
+            }
+            else
+            {
+                throw ExceptionUtilities.Unreachable();
+            }
+
+            // We only want to return base definitions that are actually callable
+            if (baseSymbol is null || baseSymbol.IsAbstract)
+            {
+                return null;
+            }
+
+            isThisAssembly = baseSymbol.ContainingAssembly == symbol.ContainingAssembly;
```

Comment:
It was surprising to me, but even when `baseSymbol` is a `PEMethodSymbol`, its still an `IMethodDefinition`, so this was the only way to tell if the base call should be a ref or a def. I was originally planning to check for `IMethodDefinition` vs `IMethodReference`